File: frontend/src/app/shared/services/department.service.ts
import { Observable } from 'rxjs';
import { DepartmentMilestones } from '../models/department-milestones.model';
import { Department } from '../models/department.model';

export abstract class DepartmentService {
  abstract getAllDepartments(): Observable<Department[]>;
  abstract getDepartmentMilestones(
    departmentId: number
  ): Observable<DepartmentMilestones>;
}

File: backend/src/main/java/com/github/performancemonitor/service/DepartmentService.java
package com.github.performancemonitor.service;

import java.util.List;

import com.github.performancemonitor.model.entity.Department;
import com.github.performancemonitor.model.entity.DepartmentMilestones;

public interface DepartmentService {

    List<Department> getAllDepartments();

    DepartmentMilestones getDepartmentMilestones(long departmentId);

}
